lib/fsi.go

```diff
@@ -240,32 +239,32 @@ func (m *FSIMethods) Write(p *FSIWriteParams, res *[]StatusItem) (err error) {
 	}
 	ctx := context.TODO()
 
-	ref, err := dsref.ParseHumanFriendly(p.Ref)
-	if err != nil {
-		return err
-	}
 	if p.Ds == nil {
 		return fmt.Errorf("dataset is required")
 	}
 
-	datasetRef := reporef.RefFromDsref(ref)
-	err = repo.CanonicalizeDatasetRef(ctx, m.inst.node.Repo, &datasetRef)
+	ref, _, err := m.inst.ParseAndResolveRef(ctx, p.Ref, "local")
```

Comment:
Before, this code was using `ParseHumanFriendly`, which guarantees that the ref is only the username/dsname, like "dustmop/my_dataset" and has no path. I don't think `ParseAndResolveRef` does the same? Meaning that path would be silently ignored instead of producing an error.